Interactive objects

AlexanderSchwarer opened the pull request:
System is set up for an interactive test object, can be applied to dynamically created objects later. Background, layout, and colour scheme cleaned up. Some visuals added when the object is picked up. There is now one grid container, for objects in the grid and the grid blocks forming the background. Objects are overlaid by setting the grid-start properties of both the objects and the background squares, and declaring/inserting the objects after the background. Background squares can be clicked while holding an object by setting the object's pointer-event: none. 
Note for future, all operations done to the object when it is picked up must me reversed when it is placed.

AlexanderSchwarer committed:
background gird elements grid position set

AlexanderSchwarer committed:
grid elements positionj set

AlexanderSchwarer committed:
Interactive test object complete
Some visuals added when the objects is picked up

blacksquaresa commented on src/Interface.css:
```diff
@@ -1,40 +1,66 @@
+:root {
+    --b: 2.34vw;
```
Use descriptive names for variables, even in CSS

blacksquaresa commented on src/Interface.css:
```diff
@@ -1,40 +1,66 @@
-    --b: 30px;
     display: grid;
     grid-template-columns: repeat(20, var(--b));
     grid-template-rows: repeat(20, var(--b));
```
Check what happens when you add more than 20 rows. Are the new rows still `--b` high? `grid-auto-rows` might help (but also might not)

AlexanderSchwarer replied:
repeat can take in a number, or have auto-fill or auto-fit. What I would want would be a minmax option as I want the grid to start out taking up the whole page. The plan is to adjust the number of rows whenever I need to add more grid blocks

blacksquaresa commented on src/Interface.ts:
Sad to see you've decided to abandon TypeScript.

AlexanderSchwarer replied:
Had a bit of irrational stress about cases where their syntax would differ, and I still wasn't comfortable with javascript. Will look at migrating soon now that I've settled more

blacksquaresa commented on src/Interface.js:
```diff
@@ -0,0 +1,73 @@
+        divSpan.style.opacity = "1";
+        holding = false;
+    }
+}
```
Strongly suggest you add `prettier` to your project at this stage. It'll get increasingly difficult to add linters and formatters as your code base grows.

blacksquaresa commented on src/Interface.js:
```diff
@@ -0,0 +1,73 @@
+
+// 2nd test object
+const ob = document.createElement("div");
+ob.className = "gridSpanTest";
```
What is an `ob`? If you mean a generic object, the standard name is `obj`. Otherwise, use a descriptive variable name.

blacksquaresa commented on src/Interface.js:
```diff
@@ -0,0 +1,73 @@
+
+divSpan.addEventListener("click", handle);
+
+function handle() {
```
A good name for a method usually includes both a verb (the action) and a noun (the target of the action). A name like 'handle' is fine if you only ever handle one thing, but will quickly need to be renamed when you need to handle two things. Even if you never handle more than one thing, when you come back to this in a year's time, you'll need to look through the code above to find out what this actually handles. If the method was named `handleGridBlockClick`, you'd know immediately what it does. 

Same for `place` below.

AlexanderSchwarer committed:
review changes:
--b changed to --blockSize
prettier formatting applied
handle function is now pickupItem
place function is now placeItem
redundant early testing code removed, including "ob"

AlexanderSchwarer committed:
review changes made
